### PR TITLE
method config validation

### PR DESCRIPTION
**Pull request by mattsooknah**
New PR for https://broadinstitute.atlassian.net/browse/DSDEEPB-661

(replaces https://github.com/broadinstitute/rawls/pull/101)

### CODE REVIEW

**Comment by tedsharpe** [comment]
We're going to defer Doug's thing about returning both an unpacked value and an error to some later story?

File: src/main/scala/org/broadinstitute/dsde/rawls/jobexec/MethodConfigResolver.scala [hunk 1]
```diff
@@ -13,74 +14,70 @@ import spray.json._
...
-    case Seq() => Failure(new EmptyResultException)
-    case Seq(null) => Failure(new EmptyResultException) // sometimes expression eval returns a Seq(null) for an empty result
-    case Seq(AttributeNull) => Failure(new EmptyResultException)
-    case Seq(single) => Success(single)
-    case _ => Failure(new MultipleResultException)
+  private def getSingleResult(seq: Seq[AttributeValue], optional: Boolean): Try[AttributeValue] = {
+    def handleEmpty = if (optional) Success(AttributeNull) else Failure(new EmptyResultException)
+    seq match {
+      case Seq() => handleEmpty
+      case Seq(null) => handleEmpty
```
Comment (tedsharpe) [inline src/main/scala/org/broadinstitute/dsde/rawls/jobexec/MethodConfigResolver.scala]: Does this really happen, or are we just hiding bugs?

Comment (mattsooknah) [inline src/main/scala/org/broadinstitute/dsde/rawls/jobexec/MethodConfigResolver.scala]: ExpressionEvaluator does sometimes return ```Seq(null)``` for a valid but empty result.

**Comment by mattsooknah** [comment]
@tedsharpe Doug says to create another story for that, just so this can get merged soon.

**Pull request merged**